Tribler/community/tunnel/tunnel_community.py

```diff
@@ -475,21 +475,22 @@ def create_circuit(self, goal_hops, ctype=CIRCUIT_TYPE_DATA, callback=None, max_
                 else:
                     self._logger.debug("Try to find a connectable node to set as required_endpoint for this circuit")
                     required_endpoint = (c.sock_addr[0], c.sock_addr[1], pubkey)
-                    if self.candidate_is_connectable(c):
-                        self._logger.debug("Valid required_endpoint found for this circuit, stop looking further")
+                    if self.candidate_is_connectable(c) and not exit_candidate.become_exit:
```

Comment:
Added a note. Because exit nodes are scarce, we should save bandwidth. In the current code it's still possible to have an exit as endpoint of a non-exiting circuit, just as a fallback when there are only exiting enabled nodes available.